workflows/differentialabundance.nf

```diff
@@ -157,8 +171,44 @@ workflow DIFFERENTIALABUNDANCE {
         ch_versions = ch_versions
             .mix(AFFY_JUSTRMA_RAW.out.versions)
 
-    }
-    else if(params.study_type == 'geo_soft_file'){
+    } else if (params.study_type == 'maxquant'){
+        // For maxquant, run proteus module to import the protein abundances
+
+        // Save contrasts file to channel
+        ch_contrasts_file = Channel.from([[exp_meta, file(params.contrasts)]])
+
+        // Split contrasts for proteus and for the later modules
+        ch_contrasts_split = ch_contrasts_file
+            .splitCsv ( header:true, sep:(params.contrasts.endsWith('tsv') ? '\t' : ','))
+            .map{ it.tail().first() }
+
+        // For proteus, extract only meta and contrast variable
+        ch_contrasts_proteus = ch_contrasts_split
+            .map{
+                tuple(
+                    exp_meta,       // meta map
+                    it.variable     // contrast variable
+                )
+            }
```

Comment:
```suggestion
        ch_contrasts_split = ch_contrasts_file
            .splitCsv ( header:true, sep:(params.contrasts.endsWith('tsv') ? '\t' : ','))
            .map{ it.tail().first() }

        // For proteus, extract only meta and contrast variable
        ch_contrasts_proteus = ch_contrasts_split
            .map{
                tuple(
                    exp_meta,       // meta map
                    it.variable     // contrast variable
                )
            }
```

So, proteus will run once for every contrast variable? How does the contrast variable impact on the quantifications?

If it does, why does the proteus module return results keyed by the `meta`, rather than the `meta2` corresponding to the contrast variable?

I think we can simplify this chunk a lot, but I need to understand what's going on better first.